### ISSUE TITLE
[link-name] Docs suggests duplicates are flagged, but they are not

### ISSUE THREAD

**Issue by connorjclark** [dequelabs/axe-core#4370]
### Product

axe-core

### Product Version

4.8.4

### Latest Version

- [X] I have tested the issue with the latest version of the product

### Issue Description

https://dequeuniversity.com/rules/axe/4.7/link-name

> Ensure that all link names are accessible. It may be possible that the inner link text is not visible to a screen reader, that **there are duplicate link labels**, or that the link is not focusable.

It's mentioned above that links with duplicate labels would fail this check. However, they are not flagged:

Given:

```html
  <a href="https://example.com">click here</a>
  <a href="https://example.com">click here</a>
  <a href="https://example.com" aria-label="hiiii">click here</a>
  <a href="https://example.com" aria-label="hiiii">click here</a>
```

link-name doesn't flag anything.

I'm not sure if it's ideal or not to actually flag duplicate label links - I can see that being more annoying than helpful in some cases. So maybe this just needs a doc edit. Otherwise, it'd be good to consider that links to the same href should be allowed to share the same label (that will help a bit with some annoying cases).

**Comment by straker** [comment]
Thanks for the issue. The docs misleading about duplicate links is being tracked in https://github.com/dequelabs/axe-core/issues/4171, so I'm going to close this as a duplicate. In terms of the label, we have another rule 
[identical-links-same-purpose](https://dequeuniversity.com/rules/axe/4.8/identical-links-same-purpose?application=RuleDescription) to verify that the same link uses the same accessible name
